Source/DafnyLanguageServer/Workspace/ChangeProcessors/Relocator.cs

```diff
@@ -88,12 +88,12 @@ CancellationToken cancellationToken
       }
 
       // Requires changeEndOffset.change.Range to be not null
-      private IEnumerable<Position> MigratePosition((TextDocumentContentChangeEvent change, Position afterChangeEndOffset) changeEndOffset, Position position) {
+      private IEnumerable<Position> MigratePosition((TextDocumentContentChangeEvent change, Position? afterChangeEndOffset) changeEndOffset, Position position) {
         if (changeEndOffset.change.Range!.Contains(position)) {
           return new List<Position> { };
         }
 
-        return new List<Position> { MigratePosition(position, changeEndOffset.change.Range, changeEndOffset.afterChangeEndOffset) };
+        return new List<Position> { MigratePosition(position, changeEndOffset.change.Range, changeEndOffset.afterChangeEndOffset!) };
```

Comment:
Looks like a dangerous API to me: you have a function that claims to accept null but then assumes it's never passed null

Reply:
Is that safe?

Reply:
Unfortunately for the eye, yes, it is safe, because I won't call this function if changeEndOffset.change.Range is null, which means that afterChangeEndOffset cannot be null either because of the invariant I'm maintaining.